File: Python/376A-lever.py
lever = input()
level = lever.find('^')
left, right = 0, 0
for k in range(level):
    left += (level - k) * int(lever[k]) if lever[k] != '=' else 0

for k in range(level + 1, len(lever)):
    right += (k - level) * int(lever[k]) if lever[k] != '=' else 0

if left == right:
    print("balance")
elif left > right:
    print("left")
else:
    print("right")

File: C++/A/376A-lever.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin >> s;
    long long level = s.find('^'), left = 0, right = 0;
    for (int k = 0; k < level; k++)
        left += ((s[k] != '=') ? (level - k) * (s[k] - '0') : 0);
    for (int k = level + 1; k < s.size(); k++)
        right += ((s[k] != '=') ? (k - level) * (s[k] - '0') : 0);
    if (left == right)
        cout << "balance";
    else if (left > right)
        cout << "left";
    else
        cout << "right";
}